infra/blueprint-test/pkg/golden/golden_test.go

```diff
@@ -133,3 +133,30 @@ func TestJSONEq(t *testing.T) {
 		})
 	}
 }
+
+func TestJSONEqs(t *testing.T) {
+	tests := []struct {
+		name    string
+		data    string
+		eqPaths []string
+		opts    []goldenFileOption
+		want    string
+	}{
+		{
+			name:    "simple",
+			data:    "{\"foo\":\"bar\",\"baz\":{\"qux\":\"quz\"},\"fizz\":\"pop\"}",
+			eqPaths: []string{"foo","baz"},
+			want:    "{\"foo\":\"bar\",\"baz\":{\"qux\":\"quz\"}}",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			os.Setenv(gfUpdateEnvVar, "true")
+			defer os.Unsetenv(gfUpdateEnvVar)
+			got := NewOrUpdate(t, tt.data, tt.opts...)
+			defer os.Remove(got.GetName())
+			got.JSONEqs(assert, utils.ParseJSONResult(t, tt.want), tt.eqPaths)
```

Comment:
Not sure I fully grokked this- is want the newer data version we are testing against? Maybe we should rename the field name?

Also should we add a test case where it does not match with golden?

Reply:
I revised the test logic, let me know if it makes more sense now.

Working on how-to include a negative test case as the `JSONPathEqs()` includes an `assert`.